db.py

```diff
@@ -101,18 +93,45 @@ def get_packages(user_pubkey=None):
     with SQL_CONNECTION() as sql:
         if user_pubkey:
             sql.execute("""
-                SELECT * FROM packages
-                WHERE escrow_pubkey IN (
-                    SELECT escrow_pubkey FROM custodians WHERE custodian_pubkey = %s)
-                OR launcher_pubkey = %s
-                OR recipient_pubkey = %s""", (user_pubkey, user_pubkey, user_pubkey))
-        else:
-            sql.execute('SELECT * FROM packages')
+            SELECT * FROM packages
+            WHERE launcher_pubkey = %s""", (user_pubkey,))
+            launched = [dict(enrich_package(row), user_role='launcher') for row in sql.fetchall()]
+            sql.execute("""
+            SELECT * FROM packages
+            WHERE recipient_pubkey = %s""", (user_pubkey,))
+            received = [dict(enrich_package(row), user_role='recipient') for row in sql.fetchall()]
+            sql.execute("""
+            SELECT * FROM packages
+            WHERE escrow_pubkey IN (
+                SELECT escrow_pubkey FROM events
+                WHERE event_type = 'couriered' AND paket_user = %s)""", (user_pubkey,))
+            couriered = [dict(enrich_package(row), user_role='courier') for row in sql.fetchall()]
+            return [
+                dict(package, custodian_pubkey=package['events'][-1]['paket_user'])
+                for package in launched + received + couriered]
+        sql.execute('SELECT * FROM packages')
         return [enrich_package(row) for row in sql.fetchall()]
 
 
-def update_custodian(escrow_pubkey, custodian_pubkey):
-    """Update a package's custodian."""
+def add_event(escrow_pubkey, user_pubkey, event_type, location):
+    """Add a package event."""
     with SQL_CONNECTION() as sql:
-        sql.execute("INSERT INTO custodians (escrow_pubkey, custodian_pubkey) VALUES (%s, %s)", (
-            escrow_pubkey, custodian_pubkey))
+        sql.execute("""
+            INSERT INTO events (event_type, location, paket_user, escrow_pubkey)
+            VALUES (%s, %s, %s, %s)
+        """, (event_type, location, user_pubkey, escrow_pubkey))
+
+
+def get_events(escrow_pubkey=None):
```

Comment:
Why is escrow_pubkey optional? When would anyone wish to get all events from the table?